extensions/wikia/AdEngine/js/wrappers/prebid.js

```diff
@@ -31,6 +31,34 @@ define('ext.wikia.adEngine.wrappers.prebid', [
 		return bids.length ? bids[0] : null;
 	}
 
+	function getWinningVideoBidBySlotName(slotName, allowedBidders) {
+		var bids;
+
+		if (!win.pbjs) {
+			return null;
+		}
+
+		bids = win.pbjs.getBidResponsesForAdUnitCode(slotName).bids || [];
+
+		return bids.filter(function (bid) {
+				var canUseThisBidder = !allowedBidders || allowedBidders.indexOf(bid.bidderCode) !== -1,
+					hasVast = bid.vastUrl || bid.vastContent;
+
+				return canUseThisBidder && hasVast;
+			})
+			.reduce(function (previousBid, currentBid) {
+				if (previousBid === null && currentBid.cpm > 0) {
+					return currentBid;
+				}
+
+				if (currentBid !== null && currentBid.cpm > previousBid.cpm) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') TypeError can be thrown as "previousBid" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2259)